lms/djangoapps/courseware/courses.py

```diff
@@ -135,7 +136,12 @@ def course_image_url(course):
         else:
             url += '/images/course_image.jpg'
     else:
-        loc = StaticContent.compute_location(course.id, course.course_image)
+        try:
+            loc = StaticContent.compute_location(course.location.course_key, course.course_image)
+        except InvalidKeyError:
+            # return empty url for split course with empty "course_image"
+            return ''
```

Comment:
I don't think you understood my last comment. It seems to me that catching the `InvalidKeyError` here is casting too wide a net; we should only be checking for cases where the `course.course_image == ''`.

I propose, leave this part of the code unchanged, but before the `else` on L137, add these lines:
```
if course.course_image == '':
    url = ''
```